// @multiversx/sdk-dapp/out/types/tokens.types.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var tokens_types_exports = {};
__export(tokens_types_exports, {
  EsdtEnumType: () => EsdtEnumType,
  NftEnumType: () => NftEnumType
});
module.exports = __toCommonJS(tokens_types_exports);
var EsdtEnumType = /* @__PURE__ */ ((EsdtEnumType2) => {
  EsdtEnumType2["FungibleESDT"] = "FungibleESDT";
  return EsdtEnumType2;
})(EsdtEnumType || {});
var NftEnumType = /* @__PURE__ */ ((NftEnumType2) => {
  NftEnumType2["NonFungibleESDT"] = "NonFungibleESDT";
  NftEnumType2["SemiFungibleESDT"] = "SemiFungibleESDT";
  NftEnumType2["MetaESDT"] = "MetaESDT";
  return NftEnumType2;
})(NftEnumType || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  EsdtEnumType,
  NftEnumType
});
