// @multiversx/sdk-dapp/out/apiCalls/transactions/getTransactionsByHashes.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var getTransactionsByHashes_exports = {};
__export(getTransactionsByHashes_exports, {
  getTransactionsByHashes: () => getTransactionsByHashes
});
module.exports = __toCommonJS(getTransactionsByHashes_exports);
var import_getServerTransactionsByHashes = require("./getServerTransactionsByHashes");
const getTransactionsByHashes = async (pendingTransactions) => {
  const hashes = pendingTransactions.map((tx) => tx.hash);
  const responseData = await (0, import_getServerTransactionsByHashes.getServerTransactionsByHashes)(hashes);
  return pendingTransactions.map((transaction) => {
    const txOnNetwork = responseData.find(
      (txResponse) => txResponse?.txHash === transaction.hash
    );
    return {
      ...transaction,
      status: txOnNetwork?.status,
      invalidTransaction: txOnNetwork == null,
      results: txOnNetwork?.results ?? [],
      previousStatus: transaction.status?.toString() || "",
      hasStatusChanged: Boolean(
        txOnNetwork && txOnNetwork.status !== transaction.status
      )
    };
  });
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getTransactionsByHashes
});
