// @multiversx/sdk-dapp/out/providers/strategies/helpers/signTransactions/helpers/getCommonData/helpers/getTxInfoByDataField.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var getTxInfoByDataField_exports = {};
__export(getTxInfoByDataField_exports, {
  getTxInfoByDataField: () => getTxInfoByDataField
});
module.exports = __toCommonJS(getTxInfoByDataField_exports);
const defaultTransactionInfo = {
  tokenId: "",
  amount: "",
  type: "",
  multiTxData: "",
  receiver: ""
};
function getTxInfoByDataField({
  data,
  multiTransactionData,
  parsedTransactionsByDataField
}) {
  if (parsedTransactionsByDataField == null) {
    return defaultTransactionInfo;
  }
  if (data in parsedTransactionsByDataField) {
    return parsedTransactionsByDataField[data];
  }
  if (multiTransactionData != null && String(multiTransactionData) in parsedTransactionsByDataField) {
    return parsedTransactionsByDataField[multiTransactionData];
  }
  return defaultTransactionInfo;
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getTxInfoByDataField
});
