// @multiversx/sdk-dapp/out/controllers/FormatAmountController/FormatAmountController.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var FormatAmountController_exports = {};
__export(FormatAmountController_exports, {
  FormatAmountController: () => FormatAmountController
});
module.exports = __toCommonJS(FormatAmountController_exports);
var import_sdkDappUtils = require("../../lib/sdkDappUtils");
class FormatAmountController {
  static getData(props) {
    const { input } = props;
    if (!input || !(0, import_sdkDappUtils.stringIsInteger)(input)) {
      return {
        isValid: false,
        label: this.getLabel(props),
        valueInteger: "0",
        valueDecimal: this.formatDecimalPart("0", props.digits)
      };
    }
    const formattedAmount = (0, import_sdkDappUtils.formatAmount)(props);
    const isValid = (0, import_sdkDappUtils.stringIsFloat)(formattedAmount);
    const [valueInteger, valueDecimal] = formattedAmount.split(".");
    return {
      isValid,
      label: this.getLabel(props),
      valueInteger: valueInteger || "0",
      valueDecimal: this.formatDecimalPart(valueDecimal, props.digits)
    };
  }
  static getLabel(props) {
    return ` ${props.token ?? props.egldLabel ?? ""}`.trimEnd();
  }
  static formatDecimalPart(decimalPart = "", digits = 2) {
    const padded = decimalPart.padEnd(digits, "0").slice(0, digits);
    return `.${padded || "0".repeat(digits)}`;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  FormatAmountController
});
