// @multiversx/sdk-dapp/out/store/slices/loginInfo/loginInfoSlice.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var loginInfoSlice_exports = {};
__export(loginInfoSlice_exports, {
  initialState: () => initialState,
  loginInfoSlice: () => loginInfoSlice
});
module.exports = __toCommonJS(loginInfoSlice_exports);
const initialState = {
  providerType: null,
  walletConnectLogin: null,
  ledgerLogin: null,
  tokenLogin: null,
  loginExpiresAt: null
};
function getTokenInfoSlice() {
  return () => initialState;
}
const loginInfoSlice = getTokenInfoSlice();
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  initialState,
  loginInfoSlice
});
