// @multiversx/sdk-dapp/out/providers/strategies/helpers/signTransactions/helpers/calculateFeeInFiat.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var calculateFeeInFiat_exports = {};
__export(calculateFeeInFiat_exports, {
  calculateFeeInFiat: () => calculateFeeInFiat
});
module.exports = __toCommonJS(calculateFeeInFiat_exports);
var import_sdkDappUtils = require("../../../../../lib/sdkDappUtils");
var import_getUsdValue = require("../../../../../utils/operations/getUsdValue");
const calculateFeeInFiat = ({
  feeLimit,
  egldPriceInUsd,
  hideEqualSign
}) => {
  const amount = (0, import_sdkDappUtils.formatAmount)({
    input: feeLimit,
    decimals: import_sdkDappUtils.DECIMALS,
    digits: import_sdkDappUtils.DIGITS,
    showLastNonZeroDecimal: true
  });
  const feeAsUsdValue = (0, import_getUsdValue.getUsdValue)({
    amount,
    usd: egldPriceInUsd,
    decimals: import_sdkDappUtils.DIGITS
  });
  if (hideEqualSign) {
    return feeAsUsdValue;
  }
  return `\u2248 ${feeAsUsdValue}`;
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  calculateFeeInFiat
});
