// @multiversx/sdk-dapp/out/providers/strategies/helpers/signTransactions/helpers/getFeeData.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var getFeeData_exports = {};
__export(getFeeData_exports, {
  getFeeData: () => getFeeData
});
module.exports = __toCommonJS(getFeeData_exports);
var import_mvx = require("../../../../../constants/mvx.constants");
var import_sdkDappUtils = require("../../../../../lib/sdkDappUtils");
var import_calculateFeeInFiat = require("./calculateFeeInFiat");
var import_calculateFeeLimit = require("./calculateFeeLimit");
const getFeeData = ({
  transaction,
  price
}) => {
  const feeLimit = (0, import_calculateFeeLimit.calculateFeeLimit)({
    from: transaction.sender.toBech32(),
    to: transaction.receiver.toBech32(),
    gasPerDataByte: String(import_mvx.GAS_PER_DATA_BYTE),
    gasPriceModifier: String(import_mvx.GAS_PRICE_MODIFIER),
    gasLimit: transaction.gasLimit.valueOf().toString(),
    gasPrice: transaction.gasPrice.valueOf().toString(),
    data: transaction.data.toString(),
    chainId: transaction.chainID.valueOf().toString()
  });
  const feeLimitFormatted = (0, import_sdkDappUtils.formatAmount)({
    input: feeLimit,
    showLastNonZeroDecimal: true
  });
  const feeInFiatLimit = price ? (0, import_calculateFeeInFiat.calculateFeeInFiat)({
    feeLimit,
    egldPriceInUsd: price,
    hideEqualSign: true
  }) : null;
  return { feeLimitFormatted, feeInFiatLimit, feeLimit };
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getFeeData
});
