// @multiversx/sdk-dapp/out/managers/internal/WalletConnectStateManager/WalletConnectStateManager.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var WalletConnectStateManager_exports = {};
__export(WalletConnectStateManager_exports, {
  WalletConnectStateManager: () => WalletConnectStateManager
});
module.exports = __toCommonJS(WalletConnectStateManager_exports);
var import_UITags = require("../../../constants/UITags.enum");
var import_UnlockPanelManager = require("../../../managers/UnlockPanelManager/UnlockPanelManager.types");
var import_walletConnect = require("../../../providers/strategies/WalletConnectProviderStrategy/types/walletConnect.types");
var import_UIBaseManager = require("../UIBaseManager/UIBaseManager");
class WalletConnectStateManager extends import_UIBaseManager.UIBaseManager {
  constructor() {
    super({
      uiDataUpdateEvent: import_walletConnect.WalletConnectEventsEnum.DATA_UPDATE,
      uiTag: import_UITags.UITagsEnum.WALLET_CONNECT
    });
    this.initialData = {
      wcURI: "",
      walletConnectDeepLink: ""
    };
    this.data = { ...this.initialData };
  }
  static getInstance() {
    if (!WalletConnectStateManager.instance) {
      WalletConnectStateManager.instance = new WalletConnectStateManager();
    }
    return WalletConnectStateManager.instance;
  }
  handleClose(options) {
    if (options?.isLoginFinished) {
      return;
    }
    if (this.anchor) {
      this.anchor.dispatchEvent(
        new CustomEvent(import_UnlockPanelManager.UnlockPanelEventsEnum.ANCHOR_CLOSE, {
          composed: false,
          bubbles: false
        })
      );
    } else {
      this.destroy();
    }
  }
  async setupEventListeners() {
    if (!this.eventBus) {
      return;
    }
    this.eventBus.subscribe(
      import_walletConnect.WalletConnectEventsEnum.CLOSE,
      this.handleClose.bind(this)
    );
    this.eventBus.subscribe(
      import_walletConnect.WalletConnectEventsEnum.UI_DISCONNECTED,
      this.destroy.bind(this)
    );
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  WalletConnectStateManager
});
