// @multiversx/sdk-dapp/out/utils/transactions/getTransactionsHistory/helpers/mapTransactionToListItem.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var mapTransactionToListItem_exports = {};
__export(mapTransactionToListItem_exports, {
  mapTransactionToListItem: () => mapTransactionToListItem
});
module.exports = __toCommonJS(mapTransactionToListItem_exports);
var import_explorerUrlBuilder = require("../../../../utils/transactions/explorerUrlBuilder");
var import_getExplorerLink = require("../../../../utils/transactions/getExplorerLink");
var import_getReceiverData = require("../../../../utils/transactions/getTransactionsHistory/helpers/getReceiverData");
var import_getTransactionAction = require("./getTransactionAction");
var import_getTransactionAmount = require("./getTransactionAmount");
var import_getTransactionAsset = require("./getTransactionAsset");
var import_getTransactionAssets = require("./getTransactionAssets");
const mapTransactionToListItem = ({
  transaction,
  address,
  explorerAddress,
  egldLabel,
  isPending = false
}) => {
  const { receiver, receiverAssets } = (0, import_getReceiverData.getReceiverData)(transaction);
  const isIncomingTransaction = address === receiver;
  const action = (0, import_getTransactionAction.getTransactionAction)({
    transaction,
    currentUserAddress: address,
    egldLabel,
    isPending
  });
  const transactionAssets = (0, import_getTransactionAssets.getTransactionAssets)({
    userIsReceiver: receiver === address,
    transaction,
    egldLabel
  });
  const hash = transaction.originalTxHash ?? transaction.txHash;
  const transactionListItem = {
    asset: (0, import_getTransactionAsset.getTransactionAsset)({
      ...transaction,
      transactionAssets,
      showDefaultState: isPending,
      status: transaction.status
    }),
    interactor: isIncomingTransaction ? transaction.sender : receiver,
    interactorAsset: isIncomingTransaction ? transaction.senderAssets?.iconSvg ?? transaction.senderAssets?.iconPng : receiverAssets?.iconSvg ?? receiverAssets?.iconPng,
    directionLabel: isIncomingTransaction ? "From" : "To",
    action,
    amount: (0, import_getTransactionAmount.getTransactionAmount)({
      transactionAssets,
      isIncomingTransaction
    }),
    hash,
    status: transaction.status,
    timestamp: transaction.timestamp,
    link: (0, import_getExplorerLink.getExplorerLink)({
      explorerAddress,
      to: import_explorerUrlBuilder.explorerUrlBuilder.transactionDetails(hash)
    })
  };
  return transactionListItem;
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  mapTransactionToListItem
});
