// @multiversx/sdk-dapp/out/constants/transactions.constants.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var transactions_constants_exports = {};
__export(transactions_constants_exports, {
  AVERAGE_TX_DURATION_MS: () => AVERAGE_TX_DURATION_MS,
  BATCH_TRANSACTIONS_ID_SEPARATOR: () => BATCH_TRANSACTIONS_ID_SEPARATOR,
  CANCEL_TRANSACTION_TOAST_DEFAULT_DURATION: () => CANCEL_TRANSACTION_TOAST_DEFAULT_DURATION,
  CANCEL_TRANSACTION_TOAST_ID: () => CANCEL_TRANSACTION_TOAST_ID,
  CROSS_SHARD_MAX_TOAST_WIDTH_PERCENT_DECREASE: () => CROSS_SHARD_MAX_TOAST_WIDTH_PERCENT_DECREASE,
  CROSS_SHARD_ROUNDS: () => CROSS_SHARD_ROUNDS,
  DEFAULT_TOAST_LIEFTIME: () => DEFAULT_TOAST_LIEFTIME,
  ERROR_SIGNING_TOAST_ID: () => ERROR_SIGNING_TOAST_ID,
  PROGRESS_INTERVAL_DURATION_MS: () => PROGRESS_INTERVAL_DURATION_MS,
  SAME_SHARD_MAX_TOAST_WIDTH_PERCENT_DECREASE: () => SAME_SHARD_MAX_TOAST_WIDTH_PERCENT_DECREASE,
  TRANSACTIONS_STATUS_DROP_INTERVAL_MS: () => TRANSACTIONS_STATUS_DROP_INTERVAL_MS,
  TRANSACTIONS_STATUS_POLLING_INTERVAL_MS: () => TRANSACTIONS_STATUS_POLLING_INTERVAL_MS
});
module.exports = __toCommonJS(transactions_constants_exports);
const CANCEL_TRANSACTION_TOAST_ID = "cancel-transaction-toast";
const ERROR_SIGNING_TOAST_ID = "error-signing-toast";
const AVERAGE_TX_DURATION_MS = 6e3;
const CROSS_SHARD_ROUNDS = 5;
const TRANSACTIONS_STATUS_POLLING_INTERVAL_MS = 90 * 1e3;
const TRANSACTIONS_STATUS_DROP_INTERVAL_MS = 10 * 60 * 1e3;
const CANCEL_TRANSACTION_TOAST_DEFAULT_DURATION = 1e4;
const BATCH_TRANSACTIONS_ID_SEPARATOR = "-";
const SAME_SHARD_MAX_TOAST_WIDTH_PERCENT_DECREASE = 75;
const CROSS_SHARD_MAX_TOAST_WIDTH_PERCENT_DECREASE = 25;
const PROGRESS_INTERVAL_DURATION_MS = 100;
const DEFAULT_TOAST_LIEFTIME = 5e3;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  AVERAGE_TX_DURATION_MS,
  BATCH_TRANSACTIONS_ID_SEPARATOR,
  CANCEL_TRANSACTION_TOAST_DEFAULT_DURATION,
  CANCEL_TRANSACTION_TOAST_ID,
  CROSS_SHARD_MAX_TOAST_WIDTH_PERCENT_DECREASE,
  CROSS_SHARD_ROUNDS,
  DEFAULT_TOAST_LIEFTIME,
  ERROR_SIGNING_TOAST_ID,
  PROGRESS_INTERVAL_DURATION_MS,
  SAME_SHARD_MAX_TOAST_WIDTH_PERCENT_DECREASE,
  TRANSACTIONS_STATUS_DROP_INTERVAL_MS,
  TRANSACTIONS_STATUS_POLLING_INTERVAL_MS
});
