// @multiversx/sdk-dapp/out/constants/UITags.enum.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var UITags_enum_exports = {};
__export(UITags_enum_exports, {
  UITagsEnum: () => UITagsEnum
});
module.exports = __toCommonJS(UITags_enum_exports);
var UITagsEnum = /* @__PURE__ */ ((UITagsEnum2) => {
  UITagsEnum2["LEDGER_CONNECT"] = "mvx-ledger-connect";
  UITagsEnum2["NOTIFICATIONS_FEED"] = "mvx-notifications-feed";
  UITagsEnum2["PENDING_TRANSACTIONS_PANEL"] = "mvx-pending-transactions-panel";
  UITagsEnum2["SIGN_TRANSACTIONS_PANEL"] = "mvx-sign-transactions-panel";
  UITagsEnum2["TOAST_LIST"] = "mvx-toast-list";
  UITagsEnum2["WALLET_CONNECT"] = "mvx-wallet-connect";
  UITagsEnum2["UNLOCK_PANEL"] = "mvx-unlock-panel";
  return UITagsEnum2;
})(UITagsEnum || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  UITagsEnum
});
