// @multiversx/sdk-dapp/out/constants/websocket.constants.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var websocket_constants_exports = {};
__export(websocket_constants_exports, {
  WebsocketConnectionStatusEnum: () => WebsocketConnectionStatusEnum,
  websocketConnection: () => websocketConnection
});
module.exports = __toCommonJS(websocket_constants_exports);
var WebsocketConnectionStatusEnum = /* @__PURE__ */ ((WebsocketConnectionStatusEnum2) => {
  WebsocketConnectionStatusEnum2["NOT_INITIALIZED"] = "not_initialized";
  WebsocketConnectionStatusEnum2["PENDING"] = "pending";
  WebsocketConnectionStatusEnum2["COMPLETED"] = "completed";
  return WebsocketConnectionStatusEnum2;
})(WebsocketConnectionStatusEnum || {});
const websocketConnection = {
  instance: null,
  status: "not_initialized" /* NOT_INITIALIZED */
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  WebsocketConnectionStatusEnum,
  websocketConnection
});
