// @multiversx/sdk-dapp/out/constants/webWalletProvider.constants.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var webWalletProvider_constants_exports = {};
__export(webWalletProvider_constants_exports, {
  WALLET_PROVIDER_CALLBACK_PARAM: () => import_sdk_web_wallet_provider.WALLET_PROVIDER_CALLBACK_PARAM,
  WALLET_PROVIDER_CALLBACK_PARAM_TX_SIGNED: () => import_sdk_web_wallet_provider.WALLET_PROVIDER_CALLBACK_PARAM_TX_SIGNED,
  WALLET_PROVIDER_CONNECT_URL: () => import_sdk_web_wallet_provider.WALLET_PROVIDER_CONNECT_URL,
  WALLET_PROVIDER_DEVNET: () => import_sdk_web_wallet_provider.WALLET_PROVIDER_DEVNET,
  WALLET_PROVIDER_DISCONNECT_URL: () => import_sdk_web_wallet_provider.WALLET_PROVIDER_DISCONNECT_URL,
  WALLET_PROVIDER_MAINNET: () => import_sdk_web_wallet_provider.WALLET_PROVIDER_MAINNET,
  WALLET_PROVIDER_SEND_TRANSACTION_URL: () => import_sdk_web_wallet_provider.WALLET_PROVIDER_SEND_TRANSACTION_URL,
  WALLET_PROVIDER_SIGN_MESSAGE_URL: () => import_sdk_web_wallet_provider.WALLET_PROVIDER_SIGN_MESSAGE_URL,
  WALLET_PROVIDER_SIGN_TRANSACTION_URL: () => import_sdk_web_wallet_provider.WALLET_PROVIDER_SIGN_TRANSACTION_URL,
  WALLET_PROVIDER_TESTNET: () => import_sdk_web_wallet_provider.WALLET_PROVIDER_TESTNET
});
module.exports = __toCommonJS(webWalletProvider_constants_exports);
var import_sdk_web_wallet_provider = require("@multiversx/sdk-web-wallet-provider");
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  WALLET_PROVIDER_CALLBACK_PARAM,
  WALLET_PROVIDER_CALLBACK_PARAM_TX_SIGNED,
  WALLET_PROVIDER_CONNECT_URL,
  WALLET_PROVIDER_DEVNET,
  WALLET_PROVIDER_DISCONNECT_URL,
  WALLET_PROVIDER_MAINNET,
  WALLET_PROVIDER_SEND_TRANSACTION_URL,
  WALLET_PROVIDER_SIGN_MESSAGE_URL,
  WALLET_PROVIDER_SIGN_TRANSACTION_URL,
  WALLET_PROVIDER_TESTNET
});
