// @multiversx/sdk-dapp/out/managers/internal/SignTransactionsStateManager/types/signTransactionsPanel.types.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var signTransactionsPanel_types_exports = {};
__export(signTransactionsPanel_types_exports, {
  SignEventsEnum: () => SignEventsEnum
});
module.exports = __toCommonJS(signTransactionsPanel_types_exports);
var SignEventsEnum = /* @__PURE__ */ ((SignEventsEnum2) => {
  SignEventsEnum2["CONFIRM"] = "CONFIRM_SIGN_TRANSACTIONS";
  SignEventsEnum2["NEXT"] = "NEXT_SIGN_TRANSACTIONS";
  SignEventsEnum2["BACK"] = "BACK_SIGN_TRANSACTIONS";
  SignEventsEnum2["DATA_UPDATE"] = "DATA_UPDATE_SIGN_TRANSACTIONS";
  SignEventsEnum2["SET_PPU"] = "SET_PPU_SIGN_TRANSACTIONS";
  SignEventsEnum2["CLOSE"] = "CLOSE_SIGN_TRANSACTIONS";
  return SignEventsEnum2;
})(SignEventsEnum || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  SignEventsEnum
});
