// @multiversx/sdk-dapp-ui/dist/web-components/dataTestIds.enum.js
var DataTestIdsEnum;
(function (DataTestIdsEnum) {
    DataTestIdsEnum["accessWalletBtn"] = "accessWalletBtn";
    DataTestIdsEnum["addressTableContainer"] = "addressTableContainer";
    DataTestIdsEnum["addressTableItem"] = "addressTableItem";
    DataTestIdsEnum["firstBtn"] = "firstBtn";
    DataTestIdsEnum["lastBtn"] = "lastBtn";
    DataTestIdsEnum["paginationItem"] = "paginationItem";
    DataTestIdsEnum["confirmBtn"] = "confirmBtn";
    DataTestIdsEnum["formatAmountComponent"] = "formatAmountComponent";
    DataTestIdsEnum["formatAmountDecimals"] = "formatAmountDecimals";
    DataTestIdsEnum["formatAmountInt"] = "formatAmountInt";
    DataTestIdsEnum["formatAmountSymbol"] = "formatAmountSymbol";
    DataTestIdsEnum["ledgerConfirmAddress"] = "ledgerConfirmAddress";
    DataTestIdsEnum["ledgerConnectBtn"] = "ledgerConnectBtn";
    DataTestIdsEnum["ledgerLoading"] = "ledgerLoading";
    DataTestIdsEnum["method"] = "method";
    DataTestIdsEnum["nextBtn"] = "nextBtn";
    DataTestIdsEnum["pendingTransactionsSubtitle"] = "pendingTransactionsSubtitle";
    DataTestIdsEnum["pendingTransactionsTitle"] = "pendingTransactionsTitle";
    DataTestIdsEnum["prevBtn"] = "prevBtn";
    DataTestIdsEnum["receiverShard"] = "receiverShard";
    DataTestIdsEnum["receiverLink"] = "receiverLink";
    DataTestIdsEnum["senderShard"] = "senderShard";
    DataTestIdsEnum["senderLink"] = "senderLink";
    DataTestIdsEnum["shardFromLink"] = "shardFromLink";
    DataTestIdsEnum["shardToLink"] = "shardToLink";
    DataTestIdsEnum["signBackBtn"] = "signBackBtn";
    DataTestIdsEnum["signCancelBtn"] = "signCancelBtn";
    DataTestIdsEnum["signNextTransactionBtn"] = "signNextTransactionBtn";
    DataTestIdsEnum["signTransactionBtn"] = "signTransactionBtn";
    DataTestIdsEnum["transactionAge"] = "transactionAge";
    DataTestIdsEnum["transactionDetailsToastBody"] = "transactionDetailsToastBody";
    DataTestIdsEnum["transactionLink"] = "transactionLink";
    DataTestIdsEnum["transactionReceiver"] = "transactionReceiver";
    DataTestIdsEnum["transactionSender"] = "transactionSender";
    DataTestIdsEnum["transactionToastContent"] = "transactionToastContent";
    DataTestIdsEnum["transactionToastTitle"] = "transactionToastTitle";
    DataTestIdsEnum["transactionNftBadge"] = "transactionNftBadge";
    DataTestIdsEnum["transactionActionFormattedAmount"] = "transactionActionFormattedAmount";
    DataTestIdsEnum["trim"] = "trim";
    DataTestIdsEnum["walletConnectLoading"] = "walletConnectLoading";
    DataTestIdsEnum["walletConnectQrCode"] = "walletConnectQrCode";
    DataTestIdsEnum["walletConnetModalTitle"] = "walletConnetModalTitle";
    DataTestIdsEnum["signTransactionsPanel"] = "signTransactionsPanel";
    DataTestIdsEnum["signTransactionsPanelContent"] = "signTransactionsPanelContent";
    DataTestIdsEnum["signTransactionsTab"] = "signTransactionsTab";
    DataTestIdsEnum["signTransactionsTabText"] = "signTransactionsTabText";
    DataTestIdsEnum["signTransactionsHeader"] = "signTransactionsHeader";
    DataTestIdsEnum["signTransactionsHeaderPager"] = "signTransactionsHeaderPager";
    DataTestIdsEnum["signTransactionsHeaderPagerPrevBtn"] = "signTransactionsHeaderPagerPrevBtn";
    DataTestIdsEnum["signTransactionsHeaderPagerNextBtn"] = "signTransactionsHeaderPagerNextBtn";
    DataTestIdsEnum["signTransactionsHeaderPagerText"] = "signTransactionsHeaderPagerText";
    DataTestIdsEnum["signTransactionsHeaderOrigin"] = "signTransactionsHeaderOrigin";
    DataTestIdsEnum["signTransactionsHeaderOriginName"] = "signTransactionsHeaderOriginName";
    DataTestIdsEnum["signTransactionsFooter"] = "signTransactionsFooter";
    DataTestIdsEnum["signTransactionsFooterButtons"] = "signTransactionsFooterButtons";
    DataTestIdsEnum["signTransactionsFooterIdentity"] = "signTransactionsFooterIdentity";
    DataTestIdsEnum["signTransactionsFooterIdentityUsername"] = "signTransactionsFooterIdentityUsername";
    DataTestIdsEnum["signTransactionsFooterIdentityAddress"] = "signTransactionsFooterIdentityAddress";
    DataTestIdsEnum["signTransactionsOverview"] = "signTransactionsOverview";
    DataTestIdsEnum["signTransactionsOverviewAmountRow"] = "signTransactionsOverviewAmountRow";
    DataTestIdsEnum["signTransactionsOverviewAmountValue"] = "signTransactionsOverviewAmountValue";
    DataTestIdsEnum["signTransactionsOverviewUsdValue"] = "signTransactionsOverviewUsdValue";
    DataTestIdsEnum["signTransactionsOverviewInteractorRow"] = "signTransactionsOverviewInteractorRow";
    DataTestIdsEnum["signTransactionsOverviewInteractorName"] = "signTransactionsOverviewInteractorName";
    DataTestIdsEnum["signTransactionsOverviewActionRow"] = "signTransactionsOverviewActionRow";
    DataTestIdsEnum["signTransactionsOverviewActionValue"] = "signTransactionsOverviewActionValue";
    DataTestIdsEnum["signTransactionsOverviewNetworkFee"] = "signTransactionsOverviewNetworkFee";
    DataTestIdsEnum["signTransactionsAdvanced"] = "signTransactionsAdvanced";
    DataTestIdsEnum["signTransactionsAdvancedGasSettings"] = "signTransactionsAdvancedGasSettings";
    DataTestIdsEnum["signTransactionsAdvancedGasPrice"] = "signTransactionsAdvancedGasPrice";
    DataTestIdsEnum["signTransactionsAdvancedGasSpeedSelector"] = "signTransactionsAdvancedGasSpeedSelector";
    DataTestIdsEnum["signTransactionsAdvancedSpeedOption"] = "signTransactionsAdvancedSpeedOption";
    DataTestIdsEnum["signTransactionsAdvancedGasLimit"] = "signTransactionsAdvancedGasLimit";
    DataTestIdsEnum["signTransactionsAdvancedData"] = "signTransactionsAdvancedData";
    DataTestIdsEnum["signTransactionsAdvancedDataDecode"] = "signTransactionsAdvancedDataDecode";
    DataTestIdsEnum["signTransactionsAdvancedDataDecodeLabel"] = "signTransactionsAdvancedDataDecodeLabel";
    DataTestIdsEnum["signTransactionsAdvancedDataDecodeOptions"] = "signTransactionsAdvancedDataDecodeOptions";
    DataTestIdsEnum["signTransactionsAdvancedDataDecodeOption"] = "signTransactionsAdvancedDataDecodeOption";
    DataTestIdsEnum["signTransactionsAdvancedDataWrapper"] = "signTransactionsAdvancedDataWrapper";
    DataTestIdsEnum["signTransactionsAdvancedDataText"] = "signTransactionsAdvancedDataText";
    DataTestIdsEnum["signTransactionsAdvancedDataHighlight"] = "signTransactionsAdvancedDataHighlight";
})(DataTestIdsEnum || (DataTestIdsEnum = {}));

export { DataTestIdsEnum as D };
