// @multiversx/sdk-dapp-ui/dist/web-components/index.js
import { g as globalScripts } from './index2.js';
export { a as getAssetPath, r as render, s as setAssetPath, b as setNonce, c as setPlatformOptions } from './index2.js';
import { MvxAddressTable } from './mvx-address-table.js';
import { MvxAnglesLeftIcon } from './mvx-angles-left-icon.js';
import { MvxAnglesRightIcon } from './mvx-angles-right-icon.js';
import { MvxArcExtensionProviderIcon } from './mvx-arc-extension-provider-icon.js';
import { MvxArrowRightIcon } from './mvx-arrow-right-icon.js';
import { MvxArrowUpRightFromSquareIcon } from './mvx-arrow-up-right-from-square-icon.js';
import { MvxArrowUpRightIcon } from './mvx-arrow-up-right-icon.js';
import { MvxBackArrowIcon } from './mvx-back-arrow-icon.js';
import { MvxBraveExtensionProviderIcon } from './mvx-brave-extension-provider-icon.js';
import { MvxButton } from './mvx-button.js';
import { MvxCheckIcon } from './mvx-check-icon.js';
import { MvxCircleExclamationIcon } from './mvx-circle-exclamation-icon.js';
import { MvxCloseIcon } from './mvx-close-icon.js';
import { MvxCopyButton } from './mvx-copy-button.js';
import { MvxCopyIcon } from './mvx-copy-icon.js';
import { MvxCustomToast } from './mvx-custom-toast.js';
import { MvxDataWithExplorerLink } from './mvx-data-with-explorer-link.js';
import { MvxDefaultTransactionIconLarge } from './mvx-default-transaction-icon-large.js';
import { MvxDefaultTransactionIconSmall } from './mvx-default-transaction-icon-small.js';
import { MvxEdgeExtensionProviderIcon } from './mvx-edge-extension-provider-icon.js';
import { MvxExplorerLink } from './mvx-explorer-link.js';
import { MvxExtensionProviderIcon } from './mvx-extension-provider-icon.js';
import { MvxFaIcon } from './mvx-fa-icon.js';
import { MvxFirefoxExtensionProviderIcon } from './mvx-firefox-extension-provider-icon.js';
import { MvxFormatAmount } from './mvx-format-amount.js';
import { MvxGenericToast } from './mvx-generic-toast.js';
import { MvxLedgerConfirm } from './mvx-ledger-confirm.js';
import { MvxLedgerConnect } from './mvx-ledger-connect.js';
import { MvxLedgerIcon } from './mvx-ledger-icon.js';
import { MvxLedgerIntro } from './mvx-ledger-intro.js';
import { MvxLedgerProviderIcon } from './mvx-ledger-provider-icon.js';
import { MvxMagnifyingGlassIcon } from './mvx-magnifying-glass-icon.js';
import { MvxMetamaskProviderIcon } from './mvx-metamask-provider-icon.js';
import { MvxMultiversxLogoIcon } from './mvx-multiversx-logo-icon.js';
import { MvxMultiversxSymbolIcon } from './mvx-multiversx-symbol-icon.js';
import { MvxNotificationsFeed } from './mvx-notifications-feed.js';
import { MvxPagination } from './mvx-pagination.js';
import { MvxPaginationEllipsis } from './mvx-pagination-ellipsis.js';
import { MvxPaginationEllipsisForm } from './mvx-pagination-ellipsis-form.js';
import { MvxPasskeyProviderIcon } from './mvx-passkey-provider-icon.js';
import { MvxPencilIcon } from './mvx-pencil-icon.js';
import { MvxPendingTransactionsPanel } from './mvx-pending-transactions-panel.js';
import { MvxPreloader } from './mvx-preloader.js';
import { MvxProviderIdleScreen } from './mvx-provider-idle-screen.js';
import { MvxSidePanel } from './mvx-side-panel.js';
import { MvxSidePanelHeader } from './mvx-side-panel-header.js';
import { MvxSidePanelSwiper } from './mvx-side-panel-swiper.js';
import { MvxSignTransactionsAdvanced } from './mvx-sign-transactions-advanced.js';
import { MvxSignTransactionsAdvancedData } from './mvx-sign-transactions-advanced-data.js';
import { MvxSignTransactionsAdvancedDataDecode } from './mvx-sign-transactions-advanced-data-decode.js';
import { MvxSignTransactionsFooter } from './mvx-sign-transactions-footer.js';
import { MvxSignTransactionsHeader } from './mvx-sign-transactions-header.js';
import { MvxSignTransactionsOverview } from './mvx-sign-transactions-overview.js';
import { MvxSignTransactionsPanel } from './mvx-sign-transactions-panel.js';
import { MvxSimpleToast } from './mvx-simple-toast.js';
import { MvxSingleAngleDownIcon } from './mvx-single-angle-down-icon.js';
import { MvxSingleAngleLeftIcon } from './mvx-single-angle-left-icon.js';
import { MvxSingleAngleRightIcon } from './mvx-single-angle-right-icon.js';
import { MvxSpinnerIcon } from './mvx-spinner-icon.js';
import { MvxToastList } from './mvx-toast-list.js';
import { MvxTooltip } from './mvx-tooltip.js';
import { MvxTransactionAccount } from './mvx-transaction-account.js';
import { MvxTransactionAccountName } from './mvx-transaction-account-name.js';
import { MvxTransactionAge } from './mvx-transaction-age.js';
import { MvxTransactionAssetIcon } from './mvx-transaction-asset-icon.js';
import { MvxTransactionDirectionBadge } from './mvx-transaction-direction-badge.js';
import { MvxTransactionHash } from './mvx-transaction-hash.js';
import { MvxTransactionIcon } from './mvx-transaction-icon.js';
import { MvxTransactionListItem } from './mvx-transaction-list-item.js';
import { MvxTransactionMethod } from './mvx-transaction-method.js';
import { MvxTransactionShards } from './mvx-transaction-shards.js';
import { MvxTransactionToast } from './mvx-transaction-toast.js';
import { MvxTransactionToastContent } from './mvx-transaction-toast-content.js';
import { MvxTransactionToastDetails } from './mvx-transaction-toast-details.js';
import { MvxTransactionToastDetailsBody } from './mvx-transaction-toast-details-body.js';
import { MvxTransactionToastProgress } from './mvx-transaction-toast-progress.js';
import { MvxTransactionValue } from './mvx-transaction-value.js';
import { MvxTransactionsTable } from './mvx-transactions-table.js';
import { MvxTriangularWarningIcon } from './mvx-triangular-warning-icon.js';
import { MvxTrim } from './mvx-trim.js';
import { MvxUnlockButton } from './mvx-unlock-button.js';
import { MvxUnlockPanel } from './mvx-unlock-panel.js';
import { MvxUnlockPanelFooter } from './mvx-unlock-panel-footer.js';
import { MvxUnlockPanelGroup } from './mvx-unlock-panel-group.js';
import { MvxUnlockProviderButton } from './mvx-unlock-provider-button.js';
import { MvxWalletConnect } from './mvx-wallet-connect.js';
import { MvxWalletConnectAppGalleryIcon } from './mvx-wallet-connect-app-gallery-icon.js';
import { MvxWalletConnectAppStoreIcon } from './mvx-wallet-connect-app-store-icon.js';
import { MvxWalletConnectDownload } from './mvx-wallet-connect-download.js';
import { MvxWalletConnectGooglePlayIcon } from './mvx-wallet-connect-google-play-icon.js';
import { MvxWalletConnectScan } from './mvx-wallet-connect-scan.js';
import { MvxWalletProviderIcon } from './mvx-wallet-provider-icon.js';
import { MvxXaliasProviderIcon } from './mvx-xalias-provider-icon.js';
import { MvxXportalDownloadQrIcon } from './mvx-xportal-download-qr-icon.js';
import { MvxXportalQrCodePreloader } from './mvx-xportal-qr-code-preloader.js';

/**
 * @fileoverview entry point for your component library
 *
 * This is the entry point for your component library. Use this file to export utilities,
 * constants or data structure that accompany your components.
 *
 * DO NOT use this file to export your components. Instead, use the recommended approaches
 * to consume components of this package as outlined in the `README.md`.
 */

globalScripts();
const defineCustomElements = (opts) => {
    if (typeof customElements !== 'undefined') {
        [
            MvxAddressTable,
            MvxAnglesLeftIcon,
            MvxAnglesRightIcon,
            MvxArcExtensionProviderIcon,
            MvxArrowRightIcon,
            MvxArrowUpRightFromSquareIcon,
            MvxArrowUpRightIcon,
            MvxBackArrowIcon,
            MvxBraveExtensionProviderIcon,
            MvxButton,
            MvxCheckIcon,
            MvxCircleExclamationIcon,
            MvxCloseIcon,
            MvxCopyButton,
            MvxCopyIcon,
            MvxCustomToast,
            MvxDataWithExplorerLink,
            MvxDefaultTransactionIconLarge,
            MvxDefaultTransactionIconSmall,
            MvxEdgeExtensionProviderIcon,
            MvxExplorerLink,
            MvxExtensionProviderIcon,
            MvxFaIcon,
            MvxFirefoxExtensionProviderIcon,
            MvxFormatAmount,
            MvxGenericToast,
            MvxLedgerConfirm,
            MvxLedgerConnect,
            MvxLedgerIcon,
            MvxLedgerIntro,
            MvxLedgerProviderIcon,
            MvxMagnifyingGlassIcon,
            MvxMetamaskProviderIcon,
            MvxMultiversxLogoIcon,
            MvxMultiversxSymbolIcon,
            MvxNotificationsFeed,
            MvxPagination,
            MvxPaginationEllipsis,
            MvxPaginationEllipsisForm,
            MvxPasskeyProviderIcon,
            MvxPencilIcon,
            MvxPendingTransactionsPanel,
            MvxPreloader,
            MvxProviderIdleScreen,
            MvxSidePanel,
            MvxSidePanelHeader,
            MvxSidePanelSwiper,
            MvxSignTransactionsAdvanced,
            MvxSignTransactionsAdvancedData,
            MvxSignTransactionsAdvancedDataDecode,
            MvxSignTransactionsFooter,
            MvxSignTransactionsHeader,
            MvxSignTransactionsOverview,
            MvxSignTransactionsPanel,
            MvxSimpleToast,
            MvxSingleAngleDownIcon,
            MvxSingleAngleLeftIcon,
            MvxSingleAngleRightIcon,
            MvxSpinnerIcon,
            MvxToastList,
            MvxTooltip,
            MvxTransactionAccount,
            MvxTransactionAccountName,
            MvxTransactionAge,
            MvxTransactionAssetIcon,
            MvxTransactionDirectionBadge,
            MvxTransactionHash,
            MvxTransactionIcon,
            MvxTransactionListItem,
            MvxTransactionMethod,
            MvxTransactionShards,
            MvxTransactionToast,
            MvxTransactionToastContent,
            MvxTransactionToastDetails,
            MvxTransactionToastDetailsBody,
            MvxTransactionToastProgress,
            MvxTransactionValue,
            MvxTransactionsTable,
            MvxTriangularWarningIcon,
            MvxTrim,
            MvxUnlockButton,
            MvxUnlockPanel,
            MvxUnlockPanelFooter,
            MvxUnlockPanelGroup,
            MvxUnlockProviderButton,
            MvxWalletConnect,
            MvxWalletConnectAppGalleryIcon,
            MvxWalletConnectAppStoreIcon,
            MvxWalletConnectDownload,
            MvxWalletConnectGooglePlayIcon,
            MvxWalletConnectScan,
            MvxWalletProviderIcon,
            MvxXaliasProviderIcon,
            MvxXportalDownloadQrIcon,
            MvxXportalQrCodePreloader,
        ].forEach(cmp => {
            if (!customElements.get(cmp.is)) {
                customElements.define(cmp.is, cmp, opts);
            }
        });
    }
};

export { defineCustomElements };
