// @multiversx/sdk-dapp-ui/dist/web-components/p-BVilz7qz.js
import { p as proxyCustomElement, H, h } from './p-CX8jKKkc.js';
import { c as classNames } from './p-DMLRPGid.js';
import { D as DataTestIdsEnum } from './p-BDDwYSuV.js';
import { d as defineCustomElement$5 } from './p-BfPNB_9u.js';
import { d as defineCustomElement$4 } from './p-prIiirvJ.js';
import { d as defineCustomElement$3 } from './p-CrqSzchw.js';
import { d as defineCustomElement$2 } from './p-DO4BP5pu.js';
import { d as defineCustomElement$1 } from './p-CdYfumVt.js';

var faFileAlt = {};

var faFileLines = {};

var hasRequiredFaFileLines;

function requireFaFileLines () {
	if (hasRequiredFaFileLines) return faFileLines;
	hasRequiredFaFileLines = 1;
	(function (exports) {
		Object.defineProperty(exports, '__esModule', { value: true });
		var prefix = 'fas';
		var iconName = 'file-lines';
		var width = 384;
		var height = 512;
		var aliases = [128441,128462,61686,"file-alt","file-text"];
		var unicode = 'f15c';
		var svgPathData = 'M0 64C0 28.7 28.7 0 64 0L213.5 0c17 0 33.3 6.7 45.3 18.7L365.3 125.3c12 12 18.7 28.3 18.7 45.3L384 448c0 35.3-28.7 64-64 64L64 512c-35.3 0-64-28.7-64-64L0 64zm208-5.5l0 93.5c0 13.3 10.7 24 24 24L325.5 176 208 58.5zM120 256c-13.3 0-24 10.7-24 24s10.7 24 24 24l144 0c13.3 0 24-10.7 24-24s-10.7-24-24-24l-144 0zm0 96c-13.3 0-24 10.7-24 24s10.7 24 24 24l144 0c13.3 0 24-10.7 24-24s-10.7-24-24-24l-144 0z';

		exports.definition = {
		  prefix: prefix,
		  iconName: iconName,
		  icon: [
		    width,
		    height,
		    aliases,
		    unicode,
		    svgPathData
		  ]};

		exports.faFileLines = exports.definition;
		exports.prefix = prefix;
		exports.iconName = iconName;
		exports.width = width;
		exports.height = height;
		exports.ligatures = aliases;
		exports.unicode = unicode;
		exports.svgPathData = svgPathData;
		exports.aliases = aliases; 
	} (faFileLines));
	return faFileLines;
}

var hasRequiredFaFileAlt;

function requireFaFileAlt () {
	if (hasRequiredFaFileAlt) return faFileAlt;
	hasRequiredFaFileAlt = 1;
	(function (exports) {
		Object.defineProperty(exports, '__esModule', { value: true });
		var source = /*@__PURE__*/ requireFaFileLines();
		exports.definition = {
		  prefix: source.prefix,
		  iconName: source.iconName,
		  icon: [
		    source.width,
		    source.height,
		    source.aliases,
		    source.unicode,
		    source.svgPathData
		  ]};

		exports.faFileAlt = exports.definition;
		exports.prefix = source.prefix;
		exports.iconName = source.iconName;
		exports.width = source.width;
		exports.height = source.height;
		exports.ligatures = source.aliases;
		exports.unicode = source.unicode;
		exports.svgPathData = source.svgPathData;
		exports.aliases = source.aliases; 
	} (faFileAlt));
	return faFileAlt;
}

var faFileAltExports = /*@__PURE__*/ requireFaFileAlt();

var faLock = {};

var hasRequiredFaLock;

function requireFaLock () {
	if (hasRequiredFaLock) return faLock;
	hasRequiredFaLock = 1;
	(function (exports) {
		Object.defineProperty(exports, '__esModule', { value: true });
		var prefix = 'fas';
		var iconName = 'lock';
		var width = 384;
		var height = 512;
		var aliases = [128274];
		var unicode = 'f023';
		var svgPathData = 'M128 96l0 64 128 0 0-64c0-35.3-28.7-64-64-64s-64 28.7-64 64zM64 160l0-64C64 25.3 121.3-32 192-32S320 25.3 320 96l0 64c35.3 0 64 28.7 64 64l0 224c0 35.3-28.7 64-64 64L64 512c-35.3 0-64-28.7-64-64L0 224c0-35.3 28.7-64 64-64z';

		exports.definition = {
		  prefix: prefix,
		  iconName: iconName,
		  icon: [
		    width,
		    height,
		    aliases,
		    unicode,
		    svgPathData
		  ]};

		exports.faLock = exports.definition;
		exports.prefix = prefix;
		exports.iconName = iconName;
		exports.width = width;
		exports.height = height;
		exports.ligatures = aliases;
		exports.unicode = unicode;
		exports.svgPathData = svgPathData;
		exports.aliases = aliases; 
	} (faLock));
	return faLock;
}

var faLockExports = /*@__PURE__*/ requireFaLock();

const transactionAccountCss = "/*! tailwindcss v4.1.3 | MIT License | https://tailwindcss.com */@layer theme{:host,:root{--mvx-color-pending:var(--mvx-pending-color);--mvx-color-error:var(--mvx-error-color);--mvx-color-primary:var(--mvx-text-color-primary);--mvx-font-sans:ui-sans-serif,system-ui,sans-serif,\"Apple Color Emoji\",\"Segoe UI Emoji\",\"Segoe UI Symbol\",\"Noto Color Emoji\";--mvx-font-mono:ui-monospace,SFMono-Regular,Menlo,Monaco,Consolas,\"Liberation Mono\",\"Courier New\",monospace;--mvx-default-font-family:var(--mvx-font-sans);--mvx-default-mono-font-family:var(--mvx-font-mono)}}@layer base{*,::backdrop,:after,:before{border:0 solid;box-sizing:border-box;margin:0;padding:0}::file-selector-button{appearance:button;background-color:#0000;border:0 solid;border-radius:0;box-sizing:border-box;color:inherit;font:inherit;font-feature-settings:inherit;font-variation-settings:inherit;letter-spacing:inherit;margin:0;margin-inline-end:4px;opacity:1;padding:0}:host,html{-webkit-text-size-adjust:100%;font-family:var(--mvx-default-font-family,ui-sans-serif,system-ui,sans-serif,\"Apple Color Emoji\",\"Segoe UI Emoji\",\"Segoe UI Symbol\",\"Noto Color Emoji\");font-feature-settings:var(--mvx-default-font-feature-settings,normal);font-variation-settings:var(--mvx-default-font-variation-settings,normal);line-height:1.5;tab-size:4;-webkit-tap-highlight-color:transparent}hr{border-top-width:1px;color:inherit;height:0}abbr:where([title]){-webkit-text-decoration:underline dotted;text-decoration:underline dotted}h1,h2,h3,h4,h5,h6{font-size:inherit;font-weight:inherit}a{color:inherit;-webkit-text-decoration:inherit;text-decoration:inherit}b,strong{font-weight:bolder}code,kbd,pre,samp{font-family:var(--mvx-default-mono-font-family,ui-monospace,SFMono-Regular,Menlo,Monaco,Consolas,\"Liberation Mono\",\"Courier New\",monospace);font-feature-settings:var(--mvx-default-mono-font-feature-settings,normal);font-size:1em;font-variation-settings:var(--mvx-default-mono-font-variation-settings,normal)}small{font-size:80%}sub,sup{font-size:75%;line-height:0;position:relative;vertical-align:baseline}sub{bottom:-.25em}sup{top:-.5em}table{border-collapse:collapse;border-color:inherit;text-indent:0}:-moz-focusring{outline:auto}progress{vertical-align:baseline}summary{display:list-item}menu,ol,ul{list-style:none}audio,canvas,embed,iframe,img,object,svg,video{display:block;vertical-align:middle}img,video{height:auto;max-width:100%}button,input,optgroup,select,textarea{background-color:#0000;border-radius:0;color:inherit;font:inherit;font-feature-settings:inherit;font-variation-settings:inherit;letter-spacing:inherit;opacity:1}:where(select:is([multiple],[size])) optgroup{font-weight:bolder}:where(select:is([multiple],[size])) optgroup option{padding-inline-start:20px}::placeholder{opacity:1}@supports (not ((-webkit-appearance:-apple-pay-button))) or (contain-intrinsic-size:1px){::placeholder{color:currentColor}@supports (color:color-mix(in lab,red,red)){::placeholder{color:color-mix(in oklab,currentColor 50%,transparent)}}}textarea{resize:vertical}::-webkit-search-decoration{-webkit-appearance:none}::-webkit-date-and-time-value{min-height:1lh;text-align:inherit}::-webkit-datetime-edit{display:inline-flex;padding-block:0}::-webkit-datetime-edit-fields-wrapper{padding:0}::-webkit-datetime-edit-month-field,::-webkit-datetime-edit-year-field{padding-block:0}::-webkit-datetime-edit-day-field,::-webkit-datetime-edit-hour-field{padding-block:0}::-webkit-datetime-edit-minute-field,::-webkit-datetime-edit-second-field{padding-block:0}::-webkit-datetime-edit-meridiem-field,::-webkit-datetime-edit-millisecond-field{padding-block:0}:-moz-ui-invalid{box-shadow:none}button,input:where([type=button],[type=reset],[type=submit]){appearance:button}::-webkit-inner-spin-button,::-webkit-outer-spin-button{height:auto}[hidden]:where(:not([hidden=until-found])){display:none!important}}@layer components;@layer utilities{.mvx\\:flex{display:flex}.mvx\\:items-center{align-items:center}.mvx\\:justify-center{justify-content:center}.mvx\\:text-error{color:var(--mvx-color-error)}.mvx\\:text-pending{color:var(--mvx-color-pending)}.mvx\\:text-primary{color:var(--mvx-color-primary)}.mvx\\:text-primary\\!{color:var(--mvx-color-primary)!important}} /*! tailwindcss v4.1.3 | MIT License | https://tailwindcss.com */@layer base{}@layer components,utilities /*! tailwindcss v4.1.3 | MIT License | https://tailwindcss.com */@layer theme{:host,:root{--mvx-color-pending:var(--mvx-pending-color);--mvx-color-error:var(--mvx-error-color);--mvx-font-sans:ui-sans-serif,system-ui,sans-serif,\"Apple Color Emoji\",\"Segoe UI Emoji\",\"Segoe UI Symbol\",\"Noto Color Emoji\";--mvx-font-mono:ui-monospace,SFMono-Regular,Menlo,Monaco,Consolas,\"Liberation Mono\",\"Courier New\",monospace;--mvx-default-font-family:var(--mvx-font-sans);--mvx-default-mono-font-family:var(--mvx-font-mono);--mvx-color-primary:var(--mvx-text-color-primary)}}@layer base{}@layer base{}@layer base{}@layer components; /*! tailwindcss v4.1.3 | MIT License | https://tailwindcss.com */@layer base{}@layer base{}@layer components; /*! tailwindcss v4.1.3 | MIT License | https://tailwindcss.com */@layer base{}@layer base{}@layer components; /*! tailwindcss v4.1.3 | MIT License | https://tailwindcss.com */@layer base{}@layer components; /*! tailwindcss v4.1.3 | MIT License | https://tailwindcss.com */@layer base{}@layer components; /*! tailwindcss v4.1.3 | MIT License | https://tailwindcss.com */@layer base{}@layer base{}@layer components; /*! tailwindcss v4.1.3 | MIT License | https://tailwindcss.com */@layer base{}@layer base{}@layer components;.transaction-account{align-items:center;display:flex;gap:.5rem}";

const transactionAccountClasses = {
    explorerLink: 'mvx:text-primary!',
};
const TransactionAccount = /*@__PURE__*/ proxyCustomElement(class TransactionAccount extends H {
    constructor() {
        super();
        this.__registerHost();
        this.showLockedAccounts = false;
    }
    render() {
        const explorerLinkDataTestId = this.scope === 'receiver' ? DataTestIdsEnum.receiverLink : DataTestIdsEnum.senderLink;
        return (h("div", { key: 'fbc5d9d7201f2e9708e9882191741a4d01ce6137', class: classNames(this.class, 'transaction-account'), "data-testid": this.dataTestId }, this.showLockedAccounts && this.account.isTokenLocked && (h("mvx-fa-icon", { key: 'f12cde9b0cd7a0d36fee714af00b03fbfdca2265', icon: faLockExports.faLock, description: this.account.name })), this.account.isContract && h("mvx-fa-icon", { key: '1924a4ce7aa2c889fa126be13f8b913a6f2c5af9', icon: faFileAltExports.faFileAlt, description: "Smart Contract" }), this.account.showLink ? (h("mvx-explorer-link", { link: this.account.link, "data-testid": explorerLinkDataTestId, class: transactionAccountClasses.explorerLink }, h("span", null, this.account.address))) : (h("mvx-transaction-account-name", { name: this.account.name, description: this.account.description, address: this.account.address }))));
    }
    static get style() { return transactionAccountCss; }
}, [256, "mvx-transaction-account", {
        "account": [16],
        "class": [1],
        "dataTestId": [1, "data-test-id"],
        "scope": [1],
        "showLockedAccounts": [4, "show-locked-accounts"]
    }]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["mvx-transaction-account", "mvx-arrow-up-right-from-square-icon", "mvx-explorer-link", "mvx-fa-icon", "mvx-transaction-account-name", "mvx-trim"];
    components.forEach(tagName => { switch (tagName) {
        case "mvx-transaction-account":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, TransactionAccount);
            }
            break;
        case "mvx-arrow-up-right-from-square-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$5();
            }
            break;
        case "mvx-explorer-link":
            if (!customElements.get(tagName)) {
                defineCustomElement$4();
            }
            break;
        case "mvx-fa-icon":
            if (!customElements.get(tagName)) {
                defineCustomElement$3();
            }
            break;
        case "mvx-transaction-account-name":
            if (!customElements.get(tagName)) {
                defineCustomElement$2();
            }
            break;
        case "mvx-trim":
            if (!customElements.get(tagName)) {
                defineCustomElement$1();
            }
            break;
    } });
}

export { TransactionAccount as T, defineCustomElement as d };
