// @multiversx/sdk-dapp-ui/dist/web-components/p-CnvVnVgY.js
import { p as proxyCustomElement, H, h } from './p-CX8jKKkc.js';

const AnglesRightIcon = /*@__PURE__*/ proxyCustomElement(class AnglesRightIcon extends H {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
    }
    render() {
        return (h("svg", { key: '5777b6049bb45a255990500e0f2426b4794ae85c', xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 512 512", width: "16", height: "16" }, h("path", { key: 'f102e6f13edb7cc2a912f6412f5fbc0d66195569', d: "M470.6 278.6c12.5-12.5 12.5-32.8 0-45.3l-160-160c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L402.7 256 265.4 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l160-160zm-352 160l160-160c12.5-12.5 12.5-32.8 0-45.3l-160-160c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L210.7 256 73.4 393.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0z" })));
    }
}, [257, "mvx-angles-right-icon"]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["mvx-angles-right-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "mvx-angles-right-icon":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, AnglesRightIcon);
            }
            break;
    } });
}

export { AnglesRightIcon as A, defineCustomElement as d };
