// @multiversx/sdk-dapp-ui/dist/web-components/p-DdrRtgNT.js
var BrowserEnum;
(function (BrowserEnum) {
    BrowserEnum["Chrome"] = "Chrome";
    BrowserEnum["Firefox"] = "Firefox";
    BrowserEnum["Edge"] = "Edge";
    BrowserEnum["Brave"] = "Brave";
    BrowserEnum["Arc"] = "Arc";
})(BrowserEnum || (BrowserEnum = {}));

const safeWindow = typeof window !== 'undefined' ? window : {};

const getDetectedBrowser = () => {
    const nav = safeWindow === null || safeWindow === void 0 ? void 0 : safeWindow.navigator;
    const userAgent = (nav === null || nav === void 0 ? void 0 : nav.userAgent) || '';
    if (/Edg/.test(userAgent)) {
        return BrowserEnum.Edge;
    }
    if (/Firefox/.test(userAgent)) {
        return BrowserEnum.Firefox;
    }
    if (nav && typeof nav.brave !== 'undefined') {
        return BrowserEnum.Brave;
    }
    if (userAgent.toLowerCase().includes('brave')) {
        return BrowserEnum.Brave;
    }
    const el = document.createElement('div');
    document.body.appendChild(el);
    const computedStyle = getComputedStyle(el);
    const hasArcPalette = computedStyle.getPropertyValue('--arc-palette-background') !== '';
    document.body.removeChild(el);
    if (hasArcPalette) {
        return BrowserEnum.Arc;
    }
    if (/Chrome/.test(userAgent)) {
        return BrowserEnum.Chrome;
    }
};

export { BrowserEnum as B, getDetectedBrowser as g, safeWindow as s };
