// @multiversx/sdk-dapp-ui/dist/web-components/p-DL2WMCqc.js
import { p as proxyCustomElement, H, h } from './p-CX8jKKkc.js';

const TriangularWarningIcon = /*@__PURE__*/ proxyCustomElement(class TriangularWarningIcon extends H {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
    }
    render() {
        return (h("svg", { key: 'b9d817e09ebf14a99aa1e483873c6c26110f1d1d', xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 512 512" }, h("path", { key: '6a0a4136d0ab2e9bdb9cd1094ed1a251f70357dc', d: "M256 32c14.2 0 27.3 7.5 34.5 19.8l216 368c7.3 12.4 7.3 27.7 .2 40.1S486.3 480 472 480L40 480c-14.3 0-27.6-7.7-34.7-20.1s-7-27.8 .2-40.1l216-368C228.7 39.5 241.8 32 256 32zm0 128c-13.3 0-24 10.7-24 24l0 112c0 13.3 10.7 24 24 24s24-10.7 24-24l0-112c0-13.3-10.7-24-24-24zm32 224a32 32 0 1 0 -64 0 32 32 0 1 0 64 0z" })));
    }
}, [257, "mvx-triangular-warning-icon"]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["mvx-triangular-warning-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "mvx-triangular-warning-icon":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, TriangularWarningIcon);
            }
            break;
    } });
}

export { TriangularWarningIcon as T, defineCustomElement as d };
