// @multiversx/sdk-dapp-ui/dist/web-components/p-DrbYhUtV.js
import { p as proxyCustomElement, H, h } from './p-CX8jKKkc.js';

const AnglesLeftIcon = /*@__PURE__*/ proxyCustomElement(class AnglesLeftIcon extends H {
    constructor() {
        super();
        this.__registerHost();
        this.__attachShadow();
    }
    render() {
        return (h("svg", { key: 'b549fead5468645e71f535d0415b4064daa10e38', xmlns: "http://www.w3.org/2000/svg", viewBox: "0 0 512 512", width: "16", height: "16" }, h("path", { key: 'ed24ab16aaab96947cc50486a5062b57650bd873', d: "M41.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.3 256 246.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160zm352-160l-160 160c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L301.3 256 438.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0z" })));
    }
}, [257, "mvx-angles-left-icon"]);
function defineCustomElement() {
    if (typeof customElements === "undefined") {
        return;
    }
    const components = ["mvx-angles-left-icon"];
    components.forEach(tagName => { switch (tagName) {
        case "mvx-angles-left-icon":
            if (!customElements.get(tagName)) {
                customElements.define(tagName, AnglesLeftIcon);
            }
            break;
    } });
}

export { AnglesLeftIcon as A, defineCustomElement as d };
