// @multiversx/sdk-dapp/out/managers/internal/PendingTransactionsStateManager/PendingTransactionsStateManager.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var PendingTransactionsStateManager_exports = {};
__export(PendingTransactionsStateManager_exports, {
  PendingTransactionsStateManager: () => PendingTransactionsStateManager
});
module.exports = __toCommonJS(PendingTransactionsStateManager_exports);
var import_UITags = require("../../../constants/UITags.enum");
var import_SidePanelBaseManager = require("../SidePanelBaseManager/SidePanelBaseManager");
var import_pendingTransactions = require("./types/pendingTransactions.types");
class PendingTransactionsStateManager extends import_SidePanelBaseManager.SidePanelBaseManager {
  constructor() {
    super({
      uiDataUpdateEvent: import_pendingTransactions.PendingTransactionsEventsEnum.DATA_UPDATE,
      uiTag: import_UITags.UITagsEnum.PENDING_TRANSACTIONS_PANEL
    });
    this.initialData = null;
    this.data = this.initialData;
  }
  static getInstance() {
    if (!PendingTransactionsStateManager.instance) {
      PendingTransactionsStateManager.instance = new PendingTransactionsStateManager();
    }
    return PendingTransactionsStateManager.instance;
  }
  isPendingTransactionsOpen() {
    return this.isOpen;
  }
  async setupEventListeners() {
    if (!this.eventBus) {
      return;
    }
    this.subscribeToEventBus(
      import_pendingTransactions.PendingTransactionsEventsEnum.CLOSE,
      this.closeUI.bind(this)
    );
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  PendingTransactionsStateManager
});
