// @multiversx/sdk-dapp/out/services/nativeAuth/methods/buildNativeAuthConfig.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var buildNativeAuthConfig_exports = {};
__export(buildNativeAuthConfig_exports, {
  buildNativeAuthConfig: () => buildNativeAuthConfig
});
module.exports = __toCommonJS(buildNativeAuthConfig_exports);
var import_getDefaultNativeAuthConfig = require("./getDefaultNativeAuthConfig");
const buildNativeAuthConfig = (config) => {
  const defaultNativeAuthConfig = (0, import_getDefaultNativeAuthConfig.getDefaultNativeAuthConfig)();
  return {
    origin: config?.origin ?? defaultNativeAuthConfig.origin,
    blockHashShard: config?.blockHashShard,
    expirySeconds: config?.expirySeconds ?? defaultNativeAuthConfig.expirySeconds,
    apiAddress: config?.apiAddress ?? defaultNativeAuthConfig.apiAddress,
    tokenExpirationToastWarningSeconds: config?.tokenExpirationToastWarningSeconds ?? defaultNativeAuthConfig.tokenExpirationToastWarningSeconds,
    extraInfo: config?.extraInfo ?? {},
    gatewayUrl: config?.gatewayUrl,
    extraRequestHeaders: config?.extraRequestHeaders ?? {}
  };
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  buildNativeAuthConfig
});
