// @multiversx/sdk-dapp/out/managers/internal/ToastManager/types/toast.types.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var toast_types_exports = {};
__export(toast_types_exports, {
  ToastEventsEnum: () => ToastEventsEnum,
  TransactionsDefaultTitles: () => TransactionsDefaultTitles
});
module.exports = __toCommonJS(toast_types_exports);
var TransactionsDefaultTitles = /* @__PURE__ */ ((TransactionsDefaultTitles2) => {
  TransactionsDefaultTitles2["success"] = "Transaction successful";
  TransactionsDefaultTitles2["received"] = "Transaction received";
  TransactionsDefaultTitles2["failed"] = "Transaction failed";
  TransactionsDefaultTitles2["pending"] = "Processing transaction";
  TransactionsDefaultTitles2["timedOut"] = "Transaction timed out";
  TransactionsDefaultTitles2["invalid"] = "Transaction invalid";
  return TransactionsDefaultTitles2;
})(TransactionsDefaultTitles || {});
var ToastEventsEnum = /* @__PURE__ */ ((ToastEventsEnum2) => {
  ToastEventsEnum2["OPEN_NOTIFICATIONS_FEED"] = "OPEN_NOTIFICATIONS_FEED";
  ToastEventsEnum2["CLOSE"] = "CLOSE_TOAST";
  ToastEventsEnum2["TRANSACTION_TOAST_DATA_UPDATE"] = "TRANSACTION_TOAST_DATA_UPDATE";
  ToastEventsEnum2["CUSTOM_TOAST_DATA_UPDATE"] = "CUSTOM_TOAST_DATA_UPDATE";
  ToastEventsEnum2["HIDE"] = "HIDE_TOAST_LIST";
  ToastEventsEnum2["SHOW"] = "SHOW_TOAST_LIST";
  return ToastEventsEnum2;
})(ToastEventsEnum || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ToastEventsEnum,
  TransactionsDefaultTitles
});
