// @multiversx/sdk-dapp/out/store/slices/network/emptyNetwork.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var emptyNetwork_exports = {};
__export(emptyNetwork_exports, {
  emptyNetwork: () => emptyNetwork
});
module.exports = __toCommonJS(emptyNetwork_exports);
const emptyNetwork = {
  id: "not-configured",
  chainId: "",
  name: "NOT CONFIGURED",
  egldLabel: "",
  decimals: "18",
  digits: "4",
  gasPerDataByte: "1500",
  walletAddress: "",
  apiAddress: "",
  explorerAddress: "",
  apiTimeout: "4000",
  roundDuration: 6e4
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  emptyNetwork
});
