// @multiversx/sdk-dapp/out/store/storage/inMemoryStorage.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var inMemoryStorage_exports = {};
__export(inMemoryStorage_exports, {
  InMemoryStorage: () => InMemoryStorage
});
module.exports = __toCommonJS(inMemoryStorage_exports);
class InMemoryStorage {
  constructor() {
    this.storage = {};
  }
  setItem(key, value) {
    this.storage[key] = value;
  }
  getItem(key) {
    return this.storage.hasOwnProperty(key) ? this.storage[key] : null;
  }
  removeItem(key) {
    delete this.storage[key];
  }
  clear() {
    this.storage = {};
  }
  get length() {
    return Object.keys(this.storage).length;
  }
  key(index) {
    const keys = Object.keys(this.storage);
    return keys[index] || null;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  InMemoryStorage
});
