// @multiversx/sdk-dapp/out/lib/sdkCore.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var sdkCore_exports = {};
__export(sdkCore_exports, {
  Address: () => import_sdk_core.Address,
  AddressComputer: () => import_sdk_core.AddressComputer,
  Message: () => import_sdk_core.Message,
  MessageComputer: () => import_sdk_core.MessageComputer,
  Transaction: () => import_sdk_core.Transaction,
  TransactionComputer: () => import_sdk_core.TransactionComputer,
  TransactionOptions: () => import_sdk_core.TransactionOptions,
  TransactionVersion: () => import_sdk_core.TransactionVersion,
  UserPublicKey: () => import_sdk_core.UserPublicKey,
  UserVerifier: () => import_sdk_core.UserVerifier
});
module.exports = __toCommonJS(sdkCore_exports);
var import_sdk_core = require("@multiversx/sdk-core");
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Address,
  AddressComputer,
  Message,
  MessageComputer,
  Transaction,
  TransactionComputer,
  TransactionOptions,
  TransactionVersion,
  UserPublicKey,
  UserVerifier
});
