// @multiversx/sdk-dapp/out/lib/sdkDappUtils.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var sdkDappUtils_exports = {};
__export(sdkDappUtils_exports, {
  DECIMALS: () => import_constants.DECIMALS,
  DIGITS: () => import_constants.DIGITS,
  ZERO: () => import_constants.ZERO,
  formatAmount: () => import_formatAmount.formatAmount,
  recommendGasPrice: () => import_recommendGasPrice.recommendGasPrice,
  stringIsFloat: () => import_stringIsFloat.stringIsFloat,
  stringIsInteger: () => import_stringIsInteger.stringIsInteger
});
module.exports = __toCommonJS(sdkDappUtils_exports);
var import_constants = require("@multiversx/sdk-dapp-utils/out/constants");
var import_stringIsInteger = require("@multiversx/sdk-dapp-utils/out/helpers/stringIsInteger");
var import_stringIsFloat = require("@multiversx/sdk-dapp-utils/out/helpers/stringIsFloat");
var import_formatAmount = require("@multiversx/sdk-dapp-utils/out/helpers/formatAmount");
var import_recommendGasPrice = require("@multiversx/sdk-dapp-utils/out/helpers/recommendGasPrice");
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  DECIMALS,
  DIGITS,
  ZERO,
  formatAmount,
  recommendGasPrice,
  stringIsFloat,
  stringIsInteger
});
