// @multiversx/sdk-dapp/out/store/store.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var store_exports = {};
__export(store_exports, {
  createDAppStore: () => createDAppStore,
  getState: () => getState,
  getStore: () => getStore,
  initStore: () => initStore,
  setDAppStore: () => setDAppStore
});
module.exports = __toCommonJS(store_exports);
var import_middleware = require("zustand/middleware");
var import_immer = require("zustand/middleware/immer");
var import_vanilla = require("zustand/vanilla");
var import_middleware2 = require("./middleware");
var import_slices = require("./slices");
var import_transactions = require("./slices/transactions");
var import_storage = require("./storage");
const createDAppStore = (getStorageCallback) => {
  const store2 = (0, import_vanilla.createStore)(
    (0, import_middleware.devtools)(
      (0, import_middleware.persist)(
        (0, import_immer.immer)((...args) => ({
          network: (0, import_slices.networkSlice)(...args),
          account: (0, import_slices.accountSlice)(...args),
          loginInfo: (0, import_slices.loginInfoSlice)(...args),
          config: (0, import_slices.configSlice)(...args),
          transactions: (0, import_transactions.transactionsSlice)(...args),
          toasts: (0, import_slices.toastSlice)(...args),
          cache: (0, import_slices.cacheSlice)(...args),
          ui: (0, import_slices.uiSlice)(...args)
        })),
        {
          name: "sdk-dapp-store",
          storage: (0, import_middleware.createJSONStorage)(getStorageCallback)
        }
      )
    )
  );
  store2.subscribe(import_middleware2.applyMiddlewares);
  return store2;
};
let store;
const setDAppStore = (_store) => {
  store = _store;
};
const getStore = () => {
  if (!store) {
    setDAppStore(createDAppStore(() => new import_storage.InMemoryStorage()));
  }
  return store;
};
const initStore = (getStorageCallback = import_storage.defaultStorageCallback) => {
  const dappStore = createDAppStore(getStorageCallback);
  setDAppStore(dappStore);
  return dappStore;
};
const getState = () => getStore().getState();
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  createDAppStore,
  getState,
  getStore,
  initStore,
  setDAppStore
});
