// @multiversx/sdk-dapp/out/managers/NotificationsFeedManager/types/notifications.types.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var notifications_types_exports = {};
__export(notifications_types_exports, {
  NotificationsFeedEventsEnum: () => NotificationsFeedEventsEnum
});
module.exports = __toCommonJS(notifications_types_exports);
var NotificationsFeedEventsEnum = /* @__PURE__ */ ((NotificationsFeedEventsEnum2) => {
  NotificationsFeedEventsEnum2["CLOSE"] = "CLOSE_NOTIFICATIONS_FEED";
  NotificationsFeedEventsEnum2["CLEAR"] = "CLEAR_NOTIFICATIONS_FEED_HISTORY";
  NotificationsFeedEventsEnum2["OPEN"] = "OPEN_NOTIFICATIONS_FEED";
  NotificationsFeedEventsEnum2["PENDING_TRANSACTIONS_UPDATE"] = "PENDING_TRANSACTIONS_UPDATE";
  NotificationsFeedEventsEnum2["TRANSACTIONS_HISTORY_UPDATE"] = "TRANSACTIONS_HISTORY_UPDATE";
  return NotificationsFeedEventsEnum2;
})(NotificationsFeedEventsEnum || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  NotificationsFeedEventsEnum
});
