// @multiversx/sdk-dapp/out/managers/UnlockPanelManager/UnlockPanelManager.types.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var UnlockPanelManager_types_exports = {};
__export(UnlockPanelManager_types_exports, {
  UnlockPanelEventsEnum: () => UnlockPanelEventsEnum
});
module.exports = __toCommonJS(UnlockPanelManager_types_exports);
var UnlockPanelEventsEnum = /* @__PURE__ */ ((UnlockPanelEventsEnum2) => {
  UnlockPanelEventsEnum2["OPEN"] = "OPEN";
  UnlockPanelEventsEnum2["CLOSE"] = "CLOSE";
  UnlockPanelEventsEnum2["LOGIN"] = "LOGIN";
  UnlockPanelEventsEnum2["CANCEL_LOGIN"] = "CANCEL_LOGIN";
  UnlockPanelEventsEnum2["CANCEL_IN_PROVIDER"] = "CANCEL_IN_PROVIDER";
  UnlockPanelEventsEnum2["ANCHOR_CLOSE"] = "ANCHOR_CLOSE";
  return UnlockPanelEventsEnum2;
})(UnlockPanelEventsEnum || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  UnlockPanelEventsEnum
});
