// @multiversx/sdk-dapp/out/store/selectors/loginInfoSelectors.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var loginInfoSelectors_exports = {};
__export(loginInfoSelectors_exports, {
  ledgerLoginSelector: () => ledgerLoginSelector,
  loginExpiresAtSelector: () => loginExpiresAtSelector,
  loginInfoSelector: () => loginInfoSelector,
  providerTypeSelector: () => providerTypeSelector,
  tokenLoginSelector: () => tokenLoginSelector,
  walletConnectLoginSelector: () => walletConnectLoginSelector
});
module.exports = __toCommonJS(loginInfoSelectors_exports);
const loginInfoSelector = ({ loginInfo }) => loginInfo;
const tokenLoginSelector = ({ loginInfo }) => loginInfo.tokenLogin;
const walletConnectLoginSelector = ({ loginInfo }) => loginInfo.walletConnectLogin;
const providerTypeSelector = ({ loginInfo }) => loginInfo.providerType;
const ledgerLoginSelector = ({ loginInfo }) => loginInfo.ledgerLogin;
const loginExpiresAtSelector = ({ loginInfo }) => loginInfo.loginExpiresAt;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ledgerLoginSelector,
  loginExpiresAtSelector,
  loginInfoSelector,
  providerTypeSelector,
  tokenLoginSelector,
  walletConnectLoginSelector
});
