// @multiversx/sdk-dapp/out/store/selectors/networkSelectors.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var networkSelectors_exports = {};
__export(networkSelectors_exports, {
  chainIdSelector: () => chainIdSelector,
  networkConfigSelector: () => networkConfigSelector,
  networkSelector: () => networkSelector,
  roundDurationSelectorSelector: () => roundDurationSelectorSelector,
  walletAddressSelector: () => walletAddressSelector
});
module.exports = __toCommonJS(networkSelectors_exports);
const networkConfigSelector = ({ network }) => network;
const networkSelector = ({ network }) => network.network;
const chainIdSelector = ({ network: { network } }) => network.chainId;
const walletAddressSelector = ({ network: { network } }) => network.walletAddress;
const roundDurationSelectorSelector = ({
  network: { network }
}) => network.roundDuration;
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  chainIdSelector,
  networkConfigSelector,
  networkSelector,
  roundDurationSelectorSelector,
  walletAddressSelector
});
