// @multiversx/sdk-dapp/out/providers/strategies/ExtensionProviderStrategy/ExtensionProviderStrategy.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var ExtensionProviderStrategy_exports = {};
__export(ExtensionProviderStrategy_exports, {
  ExtensionProviderStrategy: () => ExtensionProviderStrategy
});
module.exports = __toCommonJS(ExtensionProviderStrategy_exports);
var import_extensionProvider = require("@multiversx/sdk-extension-provider/out/extensionProvider");
var import_providerFactory = require("../../../constants/providerFactory.constants");
var import_providerFactory2 = require("../../../providers/types/providerFactory.types");
var import_provider = require("../../../types/provider.types");
var import_BaseProviderStrategy = require("../BaseProviderStrategy/BaseProviderStrategy");
var import_signMessage = require("../helpers/signMessage/signMessage");
class ExtensionProviderStrategy extends import_BaseProviderStrategy.BaseProviderStrategy {
  constructor(address) {
    super(address);
    this.cancelAction = async () => {
      this.provider.cancelAction();
    };
    this.signTransactions = async (transactions) => {
      if (!this.provider) {
        throw new Error(import_provider.ProviderErrorsEnum.notInitialized);
      }
      const { manager, onClose } = await this.initSignState();
      try {
        const signedTransactions = await this.provider.signTransactions(transactions) ?? [];
        return signedTransactions;
      } catch (error) {
        await onClose({ shouldCancelAction: false });
        throw error;
      } finally {
        manager.closeUI();
      }
    };
    this.signMessage = async (message) => {
      if (!this.provider) {
        throw new Error(import_provider.ProviderErrorsEnum.notInitialized);
      }
      const signedMessage = await (0, import_signMessage.signMessage)({
        message,
        handleSignMessage: this.provider.signMessage.bind(this.provider),
        cancelAction: this.cancelAction,
        providerType: import_providerFactory.providerLabels.extension
      });
      return signedMessage;
    };
    this.provider = import_extensionProvider.ExtensionProvider.getInstance();
    this._login = this.provider.login.bind(this.provider);
  }
  async init() {
    this.initializeAddress();
    return this.initializeProvider();
  }
  async initializeProvider() {
    const initialized = await this.provider.init();
    if (this.address) {
      this.setAccount({ address: this.address });
    }
    return initialized;
  }
  getAddress() {
    return this.provider.getAddress();
  }
  setAccount(account) {
    return this.provider.setAccount(account);
  }
  isInitialized() {
    return this.provider.isInitialized();
  }
  logout() {
    return this.provider.logout();
  }
  getType() {
    return import_providerFactory2.ProviderTypeEnum.extension;
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ExtensionProviderStrategy
});
