// @multiversx/sdk-dapp/out/utils/window/getWindowLocation.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var getWindowLocation_exports = {};
__export(getWindowLocation_exports, {
  getWindowLocation: () => getWindowLocation
});
module.exports = __toCommonJS(getWindowLocation_exports);
var import_isWindowAvailable = require("./isWindowAvailable");
const getWindowLocation = () => {
  let isAvailable;
  try {
    isAvailable = (0, import_isWindowAvailable.isWindowAvailable)();
  } catch (_error) {
    isAvailable = false;
  }
  if (!isAvailable) {
    return {
      pathname: "",
      hash: "",
      origin: "",
      href: "",
      search: ""
    };
  }
  const {
    location: { pathname, hash, origin, href, search }
  } = window;
  return {
    pathname,
    hash,
    origin,
    href,
    search
  };
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  getWindowLocation
});
