// @multiversx/sdk-dapp/out/apiCalls/endpoints.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var endpoints_exports = {};
__export(endpoints_exports, {
  ACCOUNTS_ENDPOINT: () => ACCOUNTS_ENDPOINT,
  ADDRESS_ENDPOINT: () => ADDRESS_ENDPOINT,
  BLOCKS_ENDPOINT: () => BLOCKS_ENDPOINT,
  CODE_ENDPOINT: () => CODE_ENDPOINT,
  COLLECTIONS_ENDPOINT: () => COLLECTIONS_ENDPOINT,
  CONFIG_ENDPOINT: () => CONFIG_ENDPOINT,
  CONTRACTS_ENDPOINT: () => CONTRACTS_ENDPOINT,
  ECONOMICS_ENDPOINT: () => ECONOMICS_ENDPOINT,
  IDENTITIES_ENDPOINT: () => IDENTITIES_ENDPOINT,
  KEYS_ENDPOINT: () => KEYS_ENDPOINT,
  LOCKED_ACCOUNTS_ENDPOINT: () => LOCKED_ACCOUNTS_ENDPOINT,
  LOGS_ENDPOINT: () => LOGS_ENDPOINT,
  MINIBLOCKS_ENDPOINT: () => MINIBLOCKS_ENDPOINT,
  NETWORK_CONFIG_ENDPOINT: () => NETWORK_CONFIG_ENDPOINT,
  NFTS_ENDPOINT: () => NFTS_ENDPOINT,
  NODES_ENDPOINT: () => NODES_ENDPOINT,
  PROVIDERS_ENDPOINT: () => PROVIDERS_ENDPOINT,
  ROLES_ENDPOINT: () => ROLES_ENDPOINT,
  SC_RESULTS_ENDPOINT: () => SC_RESULTS_ENDPOINT,
  STAKE_ENDPOINT: () => STAKE_ENDPOINT,
  TOKENS_ENDPOINT: () => TOKENS_ENDPOINT,
  TRANSACTIONS_BATCH: () => TRANSACTIONS_BATCH,
  TRANSACTIONS_COUNT_ENDPOINT: () => TRANSACTIONS_COUNT_ENDPOINT,
  TRANSACTIONS_ENDPOINT: () => TRANSACTIONS_ENDPOINT,
  TRANSFERS_ENDPOINT: () => TRANSFERS_ENDPOINT
});
module.exports = __toCommonJS(endpoints_exports);
const ACCOUNTS_ENDPOINT = "accounts";
const ADDRESS_ENDPOINT = "address";
const BLOCKS_ENDPOINT = "blocks";
const CODE_ENDPOINT = "code";
const COLLECTIONS_ENDPOINT = "collections";
const CONFIG_ENDPOINT = "dapp/config";
const CONTRACTS_ENDPOINT = "contracts";
const ECONOMICS_ENDPOINT = "economics";
const IDENTITIES_ENDPOINT = "identities";
const KEYS_ENDPOINT = "keys";
const LOCKED_ACCOUNTS_ENDPOINT = "locked-accounts";
const LOGS_ENDPOINT = "logs";
const MINIBLOCKS_ENDPOINT = "miniblocks";
const NETWORK_CONFIG_ENDPOINT = "network/config";
const NFTS_ENDPOINT = "nfts";
const NODES_ENDPOINT = "nodes";
const PROVIDERS_ENDPOINT = "providers";
const ROLES_ENDPOINT = "roles";
const SC_RESULTS_ENDPOINT = "sc-results";
const STAKE_ENDPOINT = "stake";
const TOKENS_ENDPOINT = "tokens";
const TRANSACTIONS_COUNT_ENDPOINT = "transactions/count";
const TRANSACTIONS_BATCH = "batch";
const TRANSACTIONS_ENDPOINT = "transactions";
const TRANSFERS_ENDPOINT = "transfers";
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  ACCOUNTS_ENDPOINT,
  ADDRESS_ENDPOINT,
  BLOCKS_ENDPOINT,
  CODE_ENDPOINT,
  COLLECTIONS_ENDPOINT,
  CONFIG_ENDPOINT,
  CONTRACTS_ENDPOINT,
  ECONOMICS_ENDPOINT,
  IDENTITIES_ENDPOINT,
  KEYS_ENDPOINT,
  LOCKED_ACCOUNTS_ENDPOINT,
  LOGS_ENDPOINT,
  MINIBLOCKS_ENDPOINT,
  NETWORK_CONFIG_ENDPOINT,
  NFTS_ENDPOINT,
  NODES_ENDPOINT,
  PROVIDERS_ENDPOINT,
  ROLES_ENDPOINT,
  SC_RESULTS_ENDPOINT,
  STAKE_ENDPOINT,
  TOKENS_ENDPOINT,
  TRANSACTIONS_BATCH,
  TRANSACTIONS_COUNT_ENDPOINT,
  TRANSACTIONS_ENDPOINT,
  TRANSFERS_ENDPOINT
});
