// @multiversx/sdk-dapp/out/managers/internal/LedgerConnectStateManager/types/ledgerConnectEvents.enum.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var ledgerConnectEvents_enum_exports = {};
__export(ledgerConnectEvents_enum_exports, {
  LedgerConnectEventsEnum: () => LedgerConnectEventsEnum
});
module.exports = __toCommonJS(ledgerConnectEvents_enum_exports);
var LedgerConnectEventsEnum = /* @__PURE__ */ ((LedgerConnectEventsEnum2) => {
  LedgerConnectEventsEnum2["CONNECT_DEVICE"] = "CONNECT_DEVICE";
  LedgerConnectEventsEnum2["ACCESS_WALLET"] = "ACCESS_WALLET";
  LedgerConnectEventsEnum2["GO_TO_PAGE"] = "GO_TO_PAGE";
  LedgerConnectEventsEnum2["CLOSE"] = "CLOSE";
  LedgerConnectEventsEnum2["DATA_UPDATE"] = "DATA_UPDATE";
  LedgerConnectEventsEnum2["UI_DISCONNECTED"] = "UI_DISCONNECTED";
  return LedgerConnectEventsEnum2;
})(LedgerConnectEventsEnum || {});
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  LedgerConnectEventsEnum
});
