// @multiversx/sdk-dapp/out/managers/internal/SidePanelBaseManager/SidePanelBaseManager.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var SidePanelBaseManager_exports = {};
__export(SidePanelBaseManager_exports, {
  SidePanelBaseManager: () => SidePanelBaseManager
});
module.exports = __toCommonJS(SidePanelBaseManager_exports);
var import_uiActions = require("../../../store/actions/ui/uiActions");
var import_UIBaseManager = require("../UIBaseManager/UIBaseManager");
class SidePanelBaseManager extends import_UIBaseManager.UIBaseManager {
  constructor({
    uiDataUpdateEvent,
    uiTag
  }) {
    super({
      uiDataUpdateEvent,
      uiTag
    });
    this.isOpen = false;
    this.data = this.getInitialData();
  }
  async openUI() {
    if (this.isOpen && this.uiElement) {
      return;
    }
    await super.init();
    this.setIsSidePanelOpen(true);
  }
  destroy() {
    this.setIsSidePanelOpen(false);
    if (this.uiElement?.closeWithAnimation) {
      this.uiElement?.closeWithAnimation?.().finally(() => {
        super.destroy();
      });
      return;
    }
    super.destroy();
  }
  closeUI() {
    this.resetData();
    this.destroy();
  }
  setIsSidePanelOpen(isOpen) {
    this.isOpen = isOpen;
    (0, import_uiActions.setIsSidePanelOpen)(isOpen);
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  SidePanelBaseManager
});
