// @multiversx/sdk-dapp/out/utils/transactions/explorerUrlBuilder.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var explorerUrlBuilder_exports = {};
__export(explorerUrlBuilder_exports, {
  explorerUrlBuilder: () => explorerUrlBuilder
});
module.exports = __toCommonJS(explorerUrlBuilder_exports);
var import_endpoints = require("../../apiCalls/endpoints");
const explorerUrlBuilder = {
  shard: (shard) => `/${import_endpoints.BLOCKS_ENDPOINT}?shard=${shard}`,
  receiverShard: (shard) => `/${import_endpoints.TRANSACTIONS_ENDPOINT}?receivershard=${shard}`,
  senderShard: (shard) => `/${import_endpoints.TRANSACTIONS_ENDPOINT}?sendershard=${shard}`,
  transactionDetails: (hash) => `/${import_endpoints.TRANSACTIONS_ENDPOINT}/${hash}`,
  transactionDetailsScResults: (hash) => `/${import_endpoints.TRANSACTIONS_ENDPOINT}/${hash}/${import_endpoints.SC_RESULTS_ENDPOINT}`,
  transactionDetailsLogs: (hash) => `/${import_endpoints.TRANSACTIONS_ENDPOINT}/${hash}/${import_endpoints.LOGS_ENDPOINT}`,
  nodeDetails: (publicKey) => `/${import_endpoints.NODES_ENDPOINT}/${publicKey}`,
  accountDetails: (address) => `/${import_endpoints.ACCOUNTS_ENDPOINT}/${address}`,
  accountDetailsContractCode: (address) => `/${import_endpoints.ACCOUNTS_ENDPOINT}/${address}/${import_endpoints.CODE_ENDPOINT}`,
  accountDetailsTokens: (address) => `/${import_endpoints.ACCOUNTS_ENDPOINT}/${address}/${import_endpoints.TOKENS_ENDPOINT}`,
  accountDetailsNfts: (address) => `/${import_endpoints.ACCOUNTS_ENDPOINT}/${address}/${import_endpoints.NFTS_ENDPOINT}`,
  accountDetailsScResults: (address) => `/${import_endpoints.ACCOUNTS_ENDPOINT}/${address}/${import_endpoints.SC_RESULTS_ENDPOINT}`,
  accountDetailsContracts: (address) => `/${import_endpoints.ACCOUNTS_ENDPOINT}/${address}/${import_endpoints.CONTRACTS_ENDPOINT}`,
  identityDetails: (id) => `/${import_endpoints.IDENTITIES_ENDPOINT}/${id}`,
  tokenDetails: (tokenId) => `/${import_endpoints.TOKENS_ENDPOINT}/${tokenId}`,
  tokenDetailsAccounts: (tokenId) => `/${import_endpoints.TOKENS_ENDPOINT}/${tokenId}/${import_endpoints.ACCOUNTS_ENDPOINT}`,
  tokenDetailsLockedAccounts: (tokenId) => `/${import_endpoints.TOKENS_ENDPOINT}/${tokenId}/${import_endpoints.LOCKED_ACCOUNTS_ENDPOINT}`,
  tokenDetailsRoles: (tokenId) => `/${import_endpoints.TOKENS_ENDPOINT}/${tokenId}/${import_endpoints.ROLES_ENDPOINT}`,
  collectionDetails: (identifier) => `/${import_endpoints.COLLECTIONS_ENDPOINT}/${identifier}`,
  nftDetails: (identifier) => `/${import_endpoints.NFTS_ENDPOINT}/${identifier}`,
  providerDetails: (address) => `/${import_endpoints.PROVIDERS_ENDPOINT}/${address}`,
  providerDetailsTransactions: (address) => `/${import_endpoints.PROVIDERS_ENDPOINT}/${address}/${import_endpoints.TRANSACTIONS_ENDPOINT}`,
  miniblockDetails: (hash) => `/${import_endpoints.MINIBLOCKS_ENDPOINT}/${hash}`
};
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  explorerUrlBuilder
});
