// @multiversx/sdk-dapp/out/providers/strategies/CrossWindowProviderStrategy/CrossWindowProviderStrategy.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var CrossWindowProviderStrategy_exports = {};
__export(CrossWindowProviderStrategy_exports, {
  CrossWindowProviderStrategy: () => CrossWindowProviderStrategy
});
module.exports = __toCommonJS(CrossWindowProviderStrategy_exports);
var import_providerFactory = require("../../../constants/providerFactory.constants");
var import_sdkWebWalletCrossWindowProvider = require("../../../lib/sdkWebWalletCrossWindowProvider");
var import_providerFactory2 = require("../../../providers/types/providerFactory.types");
var import_networkSelectors = require("../../../store/selectors/networkSelectors");
var import_store = require("../../../store/store");
var import_provider = require("../../../types/provider.types");
var import_BaseProviderStrategy = require("../BaseProviderStrategy/BaseProviderStrategy");
var import_signMessage = require("../helpers/signMessage/signMessage");
var import_guardTransactions = require("../helpers/signTransactions/helpers/guardTransactions/guardTransactions");
class CrossWindowProviderStrategy extends import_BaseProviderStrategy.BaseProviderStrategy {
  constructor(config) {
    super(config?.address);
    this.cancelAction = async () => {
      this.provider.cancelAction();
    };
    this.signTransactions = async (transactions) => {
      if (!this.provider) {
        throw new Error(import_provider.ProviderErrorsEnum.notInitialized);
      }
      const { onClose, manager } = await this.initSignState();
      try {
        const signedTransactions = await this.provider.signTransactions(transactions) ?? [];
        const optionallyGuardedTransactions = await (0, import_guardTransactions.guardTransactions)(signedTransactions);
        return optionallyGuardedTransactions;
      } catch (error) {
        await onClose({ shouldCancelAction: true });
        throw error;
      } finally {
        manager.closeUI();
      }
    };
    this.signMessage = async (message) => {
      if (!this.provider) {
        throw new Error(import_provider.ProviderErrorsEnum.notInitialized);
      }
      const signedMessage = await (0, import_signMessage.signMessage)({
        message,
        handleSignMessage: this.provider.signMessage.bind(this.provider),
        cancelAction: this.cancelAction,
        providerType: import_providerFactory.providerLabels.crossWindow
      });
      return signedMessage;
    };
    this.walletAddress = config?.walletAddress;
    this.provider = import_sdkWebWalletCrossWindowProvider.CrossWindowProvider.getInstance();
    this._login = this.provider.login.bind(this.provider);
  }
  async init() {
    this.initializeAddress();
    return this.initializeProvider();
  }
  async initializeProvider() {
    const network = (0, import_networkSelectors.networkSelector)((0, import_store.getState)());
    const isProviderInitialized = await this.provider.init();
    this.provider.setWalletUrl(this.walletAddress ?? network.walletAddress);
    if (this.address) {
      this.provider.setAddress(this.address);
    }
    return isProviderInitialized;
  }
  logout() {
    return this.provider.logout();
  }
  getType() {
    return import_providerFactory2.ProviderTypeEnum.crossWindow;
  }
  getAddress() {
    if (!this.provider) {
      throw new Error(import_provider.ProviderErrorsEnum.notInitialized);
    }
    return this.provider.getAddress();
  }
  setAccount(account) {
    return this.provider.setAccount(account);
  }
  isInitialized() {
    return this.provider.isInitialized();
  }
}
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  CrossWindowProviderStrategy
});
