// @multiversx/sdk-dapp/out/store/slices/account/accountSlice.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var accountSlice_exports = {};
__export(accountSlice_exports, {
  accountSlice: () => accountSlice,
  initialState: () => initialState
});
module.exports = __toCommonJS(accountSlice_exports);
var import_emptyAccount = require("./emptyAccount");
const initialState = {
  address: "",
  websocketEvent: null,
  websocketBatchEvent: null,
  accounts: { "": import_emptyAccount.emptyAccount },
  ledgerAccount: null,
  publicKey: "",
  walletConnectAccount: null
};
function getAccountSlice() {
  return () => initialState;
}
const accountSlice = getAccountSlice();
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  accountSlice,
  initialState
});
